modules/network-api/src/main/java/org/apache/ignite/network/TrackableNetworkMessageHandler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import static org.apache.ignite.internal.tostring.IgniteToStringBuilder.includeSensitive;
+
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Trackable message handler that will log long-running messages.
+ */
+public class TrackableNetworkMessageHandler implements NetworkMessageHandler {
+    private static final IgniteLogger LOG = Loggers.forClass(TrackableNetworkMessageHandler.class);
+
+    /**
+     * If message handling takes more time, than this constant, we will log warning message with some information.
+     */
+    // TODO: I don't know which value to use here, let's discuss in the PR. On the one hand every IO should be highlighted and thus
+    // TODO: the value should be rather small, on the other hand, currently we have lots of slow messages and we will just flood the log
+    // TODO: will log running ones. So for now, I believe we should set rather big value here in order to ease test failure analysis.
+    private static final int WATCH_EVENT_PROCESSING_LOG_THRESHOLD_NANOS = 1_000_000;
```

Comment:
Please consider renaming the constant. It has nothing to do with watch processing

Reply:
1 million nanos is a millisecond. It would have been nice to process everything within such time, but I suggest being more pessimistic just in case

Reply:
> but I suggest being more pessimistic just in case

Any suggestions?

Reply:
We could experiment with it. The goal is to see problems, but not to pollute the log. I don't have specific numbers, unfortunately. We would need to run some suites with different values (ideally) and see the real output

Reply:
Adjusted to 5 ms for know. Suggestions still welcome.